scripts/cleanup_edomain_templates.py

```diff
@@ -100,25 +105,36 @@ def get_edomain_path(api, edomain):
             edomain_conn.get_address())
 
 
-def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass):
+def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass, provider_obj):
     """Cleanup all the empty directories on the edomain/edomain_id/master/vms
     else api calls will result in 400 Error with ovf not found,
-
     Args:
         path: path for vms directory on edomain.
         edomain: Export domain of chosen RHEVM provider.
         edomainip: edomainip to connect through ssh.
         sshname: edomain ssh credentials.
         sshpass: edomain ssh credentials.
+        provider_obj: provider object under execution
     """
     try:
-        print("RHEVM: Deleting the empty directories on edomain/vms file...")
-        ssh_client = make_ssh_client(edomainip, sshname, sshpass)
-        command = 'cd {}/master/vms && find . -maxdepth 1 -type d -empty -delete'.format(path)
+        provider_ip = provider_obj.kwargs.get('ipaddress', None)
+        provider_name = provider_obj.kwargs.get('name', None)
+        ssh_client = make_ssh_client(provider_ip, sshname, sshpass)
```

Comment:
Shouldn't we already have access to ssh_client for a provider under ```provider_obj.ssh_client``` or is this object just a dict? if it is just the dict of data, we should rename it to provider_dict.